src/Data/Generics/Product/Fields.hs

```diff
@@ -159,6 +168,18 @@ instance  -- see Note [Changing type parameters]
 instance {-# OVERLAPPING #-} HasField f (Void1 a) (Void1 b) a b where
   field = undefined
 
+instance
+  ( Generic s
+  , ErrorUnless field s (CollectField field (Rep s))
+  , Generic t
+  , GHasKey  field (Rep s) (Rep t) a b
+  , UnifyHead s t
+  ) => HasField_ field s t a b where
```

Comment:
Should I keep the `GHasKey' field (Rep s) a` from the `HasField` instance? What do those `GHasKey'` constraints do?